### ISSUE TITLE
Lint/UselessRescue: False positive for re-raising in parent thread

### ISSUE THREAD

**Issue by franzliedke** [rubocop/rubocop#11682]
We're monkey-patching a gem to elevate its exceptions to a parent thread.

```ruby
  def patched_method
    super
  rescue ::Exception => ex # rubocop:disable Lint/RescueException
    Thread.current[:parent].raise(ex)
  end
```

--------

## Expected behavior

I believe `Lint/UselessRescue` should not complain about the `raise` method call on another thread.

## Actual behavior

```
file/with/monkeypatch.rb:18:3: W: Lint/UselessRescue: Useless rescue detected.
  rescue ::Exception => ex # rubocop:disable Lint/RescueException ...
  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

## Steps to reproduce the problem

See snippet above.

## RuboCop version

```
1.48.0 (using Parser 3.2.1.1, rubocop-ast 1.27.0, running on ruby 3.2.1) [x86_64-linux]
  - rubocop-capybara 2.17.1
  - rubocop-graphql 0.19.0
  - rubocop-performance 1.16.0
  - rubocop-rails 2.17.4
  - rubocop-rake 0.6.0
  - rubocop-rspec 2.19.0
```
